src/github.com/getlantern/flashlight/logging/logging.go

```diff
@@ -67,108 +54,122 @@ func Setup(version string, buildDate string) *rotator.SizeRotator {
-		return fmt.Errorf("Proxy never came online at %s: %q", cfg.Addr, err)
-	}
+	if cfg.Addr != lastAddr {
+		lastAddr = cfg.Addr
 
-	var cli *http.Client
-	if cli, err = util.HTTPClient(cfg.CloudConfigCA, cfg.Addr); err != nil {
-		return fmt.Errorf("Could not create proxy HTTP client %q.", err)
-	}
+		if cfg.Addr == "" {
+			log.Error("No known proxy, won't report to Loggly")
+			removeLoggly()
+			return
+		}
 
-	logglyClient.l.SetHTTPClient(cli)
+		err := waitforserver.WaitForServer("tcp", cfg.Addr, 10*time.Second)
+		if err != nil {
+			log.Errorf("Proxy never came online at %v, not logging to Loggly", cfg.Addr)
+			removeLoggly()
+			return
+		}
+
+		var client *http.Client
+		client, err = util.HTTPClient(cfg.CloudConfigCA, cfg.Addr)
+		if err != nil {
+			log.Errorf("Could not create proxied HTTP client, not logging to Loggly: %v", err)
+			removeLoggly()
+			return
+		}
+
+		log.Debugf("Sending error logs to Loggly via proxy at %v", cfg.Addr)
+
+		lang, _ := jibber_jabber.DetectLanguage()
+		logglyWriter := &logglyErrorWriter{
+			lang:            lang,
+			tz:              time.Local.String(),
+			versionToLoggly: fmt.Sprintf("%v (%v)", version, buildDate),
+			client:          loggly.New(logglyToken),
+		}
+		logglyWriter.client.SetHTTPClient(client)
+		addLoggly(logglyWriter)
+	}
+}
 
-	return nil
+func Close() error {
+	return logFile.Close()
 }
 
 // timestamped adds a timestamp to the beginning of log lines
 func timestamped(orig io.Writer) io.Writer {
 	return wfilter.LinePrepender(orig, func(w io.Writer) (int, error) {
-		return fmt.Fprintf(w, "%s - ", time.Now().In(time.UTC).Format(LogTimestampFormat))
+		return fmt.Fprintf(w, "%s - ", time.Now().In(time.UTC).Format(logTimestampFormat))
 	})
 }
 
-type logglyErrorWriter struct {
-	l *loggly.Client
+func addLoggly(logglyWriter io.Writer) {
+	golog.SetOutputs(io.MultiWriter(errorOut, logglyWriter), debugOut)
 }
 
-func (w logglyErrorWriter) Write(b []byte) (int, error) {
-	return writeToLoggly(w.l, "ERROR", string(b))
+func removeLoggly() {
+	golog.SetOutputs(errorOut, debugOut)
 }
 
-func writeToLoggly(l *loggly.Client, level string, msg string) (int, error) {
+type logglyErrorWriter struct {
+	lang            string
+	tz              string
+	versionToLoggly string
+	client          *loggly.Client
+}
+
+func (w logglyErrorWriter) Write(b []byte) (int, error) {
 	extra := map[string]string{
-		"logLevel":  level,
+		"logLevel":  "ERROR",
 		"osName":    runtime.GOOS,
 		"osArch":    runtime.GOARCH,
 		"osVersion": "",
-		"language":  lang,
+		"language":  w.lang,
 		"country":   globals.GetCountry(),
-		"timeZone":  tz,
-		"version":   versionToLoggly,
+		"timeZone":  w.tz,
+		"version":   w.versionToLoggly,
 	}
 
 	m := loggly.Message{
 		"extra":   extra,
-		"message": msg,
+		"message": string(b),
 	}
-	err := l.Send(m)
+
+	err := w.client.Send(m)
 	if err != nil {
 		return 0, err
 	}
-	return len(msg), nil
-}
-
-type nonStopWriter struct {
```

Comment:
I took out nonStopWriter because I noticed that my errors weren't making it to the console.  I replaced it with io.MultiWriter, which has the downside that it stops as soon as any of the writers had an error, but the upside is that it works and my output makes it to the console.